pycbc/inference/sampler_mcmc.py

```diff
@@ -0,0 +1,554 @@
+        """
+        # get the group to load from
+        if samples_group is None:
+            samples_group = fp.samples_group
+        # get the type of array class to use
+        if array_class is None:
+            array_class = FieldArray
+        # get the names of fields needed for the given parameters
+        possible_fields = fp[samples_group].keys()
+        loadfields = array_class.parse_parameters(parameters, possible_fields)
+        return cls._read_fields(fp, samples_group, loadfields, array_class,
+                                thin_start=thin_start,
+                                thin_interval=thin_interval, thin_end=thin_end,
+                                iteration=iteration)
+
+    @staticmethod
+    def _read_fields(fp, fields_group, fields, array_class,
+                     thin_start=None, thin_interval=None, thin_end=None,
+                     iteration=None):
+        """Base function for reading samples and likelihood stats. See
+        `read_samples` and `read_likelihood_stats` for details.
+
+        Parameters
+        -----------
+        fp : InferenceFile
+            An open file handler to read the samples from.
+        fields_group : str
+            The name of the group to retrieve the desired fields.
+        fields : list
+            The list of field names to retrieve. Must be names of groups in
+            `fp[fields_group/]`.
+        array_class : FieldArray or similar
+            The type of array to return. Must have a `from_kwargs` attribute.
+
+        For other details on keyword arguments, see `read_samples` and
+        `read_likelihood_stats`.
+
+        Returns
+        -------
+        array_class
+            An instance of the given array class populated with values
+            retrieved from the fields.
+        """
+
+        # get the slice to use
+        if iteration is not None:
+            get_index = iteration
+        else:
+            if thin_end is None:
+                # use the number of current iterations
+                thin_end = fp.niterations
+            get_index = fp.get_slice(thin_start=thin_start, thin_end=thin_end,
+                                     thin_interval=thin_interval)
+
+        # load
+        arrays = {}
+        group = fields_group + '/{name}'
+        for name in fields:
+            these_arrays = fp[group.format(name=name)][get_index]
+            arrays[name] = numpy.vstack(these_arrays)
+        return array_class.from_kwargs(**arrays)
+
+
+    @staticmethod
+    def write_acls(fp, acls):
+        # Dummy method as required by line 389 of pycbc_inference
+        #                sampler.write_acls(fp, sampler.compute_acls(fp))
+        return None
+    @classmethod
+    def compute_acls(cls, fp, start_index=None, end_index=None):
+        # Dummy method as required by line 389 of pycbc_inference
+        #                sampler.write_acls(fp, sampler.compute_acls(fp))
+        return None
```

Comment:
When you inherit from `_BaseMCMCSampler`, you should be able to use the `write_acls` and `compute_acls` from there.